Rock.Encryption/Symmetric/Credential.cs

```diff
@@ -35,6 +36,32 @@ public Credential()
         }
 
         /// <summary>
+        /// Initializes a new instance of the <see cref="Credential"/> class.
+        /// </summary>
+        /// <param name="algorithm">The <see cref="SymmetricAlgorithm"/> that will be used for a symmetric encryption or decryption operation.</param>
+        /// <param name="ivSize">The size of the initialization vector that is used to add entropy to encryption or decryption operations.</param>
+        /// <param name="key">The symmetric key returned by the <see cref="GetKey()"/> method.</param>
+        public Credential(SymmetricAlgorithm algorithm, ushort ivSize, byte[] key)
+        {
+            if (ivSize <= 0) throw new ArgumentOutOfRangeException(nameof(ivSize), $"{nameof(ivSize)} must be greater than 0");
+            if (key == null) throw new ArgumentNullException(nameof(key));
+
+            Algorithm = algorithm;
+            IVSize = ivSize;
+
+            var builder = new ConfigurationBuilder();
+            builder.AddInMemoryCollection(new Dictionary<string, string> { { "Section:Value", Convert.ToBase64String(key) } });
+
+            Key = new LateBoundConfigurationSection<IProtectedValue>
```

Comment:
seems very confusing to me that we have Key and _key but they are both different types.

I just assumed that _key was the backing field for Key, but this not correct.

Reply:
In a way, it is the backing field.  In Key, it's still a config section, in _key, it's extracted from the config section for easier use later.  If we want to discuss a better way to do this, I'm fine with it, but this is how it was before any of my changes, so it should probably be a seperate PR.

Reply:
Sounds good, thank you for addressing this.  I say leave it for now